packages/next/src/server/base-server.ts

```diff
@@ -1390,20 +973,28 @@ export default abstract class Server<
         throw err
       }
 
-      // This wasn't a request via the matched path or the invoke path, so
-      // prepare for a legacy run by removing the base path.
+      res.statusCode = 200
 
-      // ensure we strip the basePath when not using an invoke header
-      if (!useMatchedPathHeader && pathnameInfo.basePath) {
-        parsedUrl.pathname = removePathPrefix(
-          parsedUrl.pathname,
-          pathnameInfo.basePath
-        )
+      // If we aren't in minimal mode, this isn't a custom server, and we aren't
+      // in the edge runtime, we should use the handle method directly, as this
+      // has been routed by Next.js.
+      if (
+        !this.minimalMode &&
+        !this.renderOpts.customServer &&
+        process.env.NEXT_RUNTIME !== 'edge'
+      ) {
+        await this.handleCatchallRenderRequest(req, res, parsedUrl)
+      } else {
+        await this.run(req, res, parsedUrl)
       }
 
-      res.statusCode = 200
-      return await this.run(req, res, parsedUrl)
+      return
     } catch (err: any) {
+      if (err instanceof XInvokeError) {
```

Comment:
It seems weird for this error to leak outside of the adapter. Why are Invoke Errors handled any differently?

Reply:
These errors are just signals to the renderer that includes details as to which error code + status to return with. The invoke pathway right now is the only one that can emit these signals. I can remove the `XInvoke` prefix and generalize it for any request adapter.